go/enclave/events/subscription_manager.go

```diff
@@ -118,13 +119,13 @@ func (s *SubscriptionManager) EncryptLogs(logsBySubID map[uuid.UUID][]*types.Log
 // Extracts the (potential) user addresses from the topics. If there is no code associated with an address, it's a user
 // address.
 func getUserAddrs(log *types.Log, db *state.StateDB) []string {
-	var nonContractAddrs []string //nolint:prealloc
+	var userAddrs []string //nolint:prealloc
 
 	for _, topic := range log.Topics {
 		// Since addresses are 20 bytes long, while hashes are 32, only topics with 12 leading zero bytes can
 		// (potentially) be user addresses.
 		topicHex := topic.Hex()
-		if topicHex[2:len(zeroBytesHex)/2] != zeroBytesHex {
+		if topicHex[2:len(zeroBytesHex)+2] != zeroBytesHex {
```

Comment:
This was a bug, leading to everything being categorised as not-potentially-an-address. It wasn't picked up, because the fact that we weren't routing events based on subscription IDs meant we couldn't do relevancy checks in the sims. The changes to the sim validation will prevent us backsliding here.